src/jit/valuenum.cpp

```diff
@@ -5633,6 +5633,9 @@ void Compiler::fgValueNumberTree(GenTree* tree, bool evalAsgLhsInd)
     {
         // TODO-CQ: For now hardware intrinsics are not handled by value numbering to be amenable for CSE'ing.
         tree->gtVNPair.SetBoth(vnStore->VNForExpr(compCurBB, TYP_UNKNOWN));
+        // Also mutate the left-hand-side node as it is likely to be modified by an unknown intrinsic operation
+        GenTree* lhs = tree->gtOp.gtOp1;
+        lhs->gtVNPair.SetBoth(vnStore->VNForExpr(compCurBB, TYP_UNKNOWN));
```

Comment:
It's not clear how changing the VN of the addr node helps. It's the memory VN that needs to be changed (via `fgMutateGcHeap` as far as I can tell).